scripts/tick-cluster.js

```diff
@@ -429,34 +439,32 @@ function reviveProcs() {
 }
 
 function suspendProc(count) {
-    toSuspend = +count;
-
-    var suspended = [];
-    while (suspended.length < toSuspend) {
-        var rand = Math.floor(Math.random() * procs.length);
-        var proc = procs[rand];
-        if (proc.killed === null && proc.suspended === null) {
-            logMsg(proc.port, color.green('pid ' + proc.pid) + color.red(' randomly selected for sleep'));
-            process.kill(proc.proc.pid, 'SIGSTOP');
-            proc.suspended = Date.now();
-            suspended.push(proc);
-        }
+    var running = procs.filter(function (proc) { return !proc.killed && !proc.suspended; });
+    toSuspend = Math.min(+count, running.length);
+    shuffle(running);
```

Comment:
Instead of using underscore's shuffle method, you could also use `sample`: http://underscorejs.org/#sample.
```
processesToSuspend = _.sample(running, +count)
processesToSuspend.each(function suspend(proc) {
    logMsg(proc.port, color.green('pid ' + proc.pid) + color.red(' randomly selected for sleep'));
    process.kill(proc.proc.pid, 'SIGSTOP');
    proc.suspended = Date.now();
}
```